### DISCUSSION TITLE
Prisma - Works in Localhost, Fails in Github Actions

### DISCUSSION THREAD

**Discussion by sbue** [prisma/prisma#20443] [Q&A]
Hey all! Having a gnarly issue. TLDR Prisma not connecting to the db in Github Actions NextJs `getStaticProps`

Prisma client code:
```js
const DATABASE_PGBOUNCER_URL = "postgresql://postgres:***@db.***.supabase.co:6543/postgres?pgbouncer=true&connection_limit=1&pool_timeout=30"

const prisma = new PrismaClient({
  log: ['query', 'info', 'warn', 'error'],
  datasources: {
    db: {
      url: process.env.DATABASE_PGBOUNCER_URL,
    },
  },
})
```

NextJs getStaticProps that is failing
```js
export const getStaticProps: GetStaticProps = async () => {
  console.log('A')
  const xyz = await fetch('https://www.google.com/')
  console.log({ B: xyz.status })
  console.log('C')
  try {
    await prisma.$queryRaw`SELECT 1`
  } catch (err) {
    console.error(err)
  }
  console.log('D')
}
```
Here's the logs in localhost:
```
A
{ B: 200 }
C
> prisma:info Starting a postgresql pool with 1 connections in PgBouncer mode.
> prisma:query BEGIN
> prisma:query DEALLOCATE ALL
> prisma:query SELECT 1
> prisma:query COMMIT
D
```
Here's the logs in github actions
```
A
{ B: 200 }
C
> prisma:info Starting a postgresql pool with 1 connections in PgBouncer mode.
  
Static worker unexpectedly exited with code: null and signal: SIGSEGV  
> Build error occurred
  Error: Static page generation for / is still timing out after 3 attempts. See more info here 
https://nextjs.org/docs/messages/static-page-generation-timeout
```

**Comment by ludralph** [comment]
Hi @sbue 👋 

Can you try the recommendation from this Next.js page?

- https://nextjs.org/docs/messages/static-page-generation-timeout#possible-ways-to-fix-it

**Reply by sbue** [reply]
Hi @ludralph - yes I've made sure to follow all the guidelines per the article you sent

* _Make sure that there is no infinite loop during execution._ -> no infinite loop (basic script)
* _Make sure all Promises in getStaticPaths/getStaticProps resolve or reject correctly._ -> simple `await prisma.$queryRaw``
* _Avoid very long timeouts for network requests._ > simple `await prisma.$queryRaw``
* _Increase the timeout by changing the staticPageGenerationTimeout configuration option (default 60 in seconds)._ -> Done, doesn't solve the problem. If timeout is made longer (say 5 minutes, the application keeps running until timeout then throws an error)

This really seems like a problem that requires support from the Prisma side moreso than the NextJs side. How is it that it doesn't emit any debug logs and stays idle without executing the request?

**Reply by ludralph** [reply]
Hi @sbue, which Prisma version are you using? Also, can you try increasing the `connection_limit` to say 5? If possible can you provide a reproduction of this issue?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
